internal/component/discovery/dns/dns.go

```diff
@@ -70,8 +71,8 @@ func (args Arguments) Convert() dns.SDConfig {
 
 // New returns a new instance of a discovery.dns component.
 func New(opts component.Options, args Arguments) (*discovery.Component, error) {
-	return discovery.New(opts, args, func(args component.Arguments) (discovery.Discoverer, error) {
+	return discovery.New(opts, args, func(args component.Arguments) (discovery.DiscovererConfig, error) {
 		conf := args.(Arguments).Convert()
-		return dns.NewDiscovery(conf, opts.Logger), nil
+		return &conf, nil
```

Comment:
nit: There seems to be a few different syntactic patterns for discovery.New. Some call the Convert() on the return and some don't. Possibly go with one over the other for consistency?

Reply:
Will take a look to unify this where possible. I think there were some already existing differences and sth different because of the pointers.

Reply:
I have unified them now and got rid of all these similar New functions. It was quite a lot to refactor and it's now added  to this PR. We should probably try not to increase the scope more here...